cmd/provisioner-localpv/app/config.go

```diff
@@ -0,0 +1,164 @@
+/*
+Copyright 2019 The OpenEBS Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+*/
+
+package app
+
+import (
+	"fmt"
+	"path/filepath"
+	"strings"
+
+	mconfig "github.com/openebs/maya/pkg/apis/openebs.io/v1alpha1"
+	cast "github.com/openebs/maya/pkg/castemplate/v1alpha1"
+	"github.com/openebs/maya/pkg/util"
+	"github.com/pkg/errors"
+	"k8s.io/api/core/v1"
+)
+
+const (
+	KeyPVBasePath     = "BasePath"
+	KeyPVRelativePath = "RelativePath"
+	KeyPVAbsolutePath = "AbsolutePath"
+)
+
+const (
+	// BetaStorageClassAnnotation represents the beta/previous StorageClass annotation.
+	// It's currently still used and will be held for backwards compatibility
+	BetaStorageClassAnnotation = "volume.beta.kubernetes.io/storage-class"
+)
+
+//CASConfigParser creates a new CASConfigPVC struct by
+// parsing and merging the configuration provided in the PVC
+// annotation - cas.openebs.io/config with the
+// default configuration of the provisioner.
+func (p *Provisioner) CASConfigParser(pvName string, pvc *v1.PersistentVolumeClaim) (*CASConfigPVC, error) {
+
+	pvConfig := p.defaultConfig
+
+	//TODO Fetch the SC and its configuration
+	scName := GetStorageClassName(pvc)
+
+	// extract the cas volume config from pvc
+	pvcCASConfigStr := pvc.ObjectMeta.Annotations[string(mconfig.CASConfigKey)]
+	if len(strings.TrimSpace(pvcCASConfigStr)) != 0 {
+		pvcCASConfig, err := cast.UnMarshallToConfig(pvcCASConfigStr)
+		if err == nil {
+			pvConfig = cast.MergeConfig(pvcCASConfig, pvConfig)
+		}
+	}
+
+	pvConfigMap, err := cast.ConfigToMap(pvConfig)
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to read configuration for pvc %v", pvc.ObjectMeta.Name)
+	}
+
+	c := &CASConfigPVC{
+		pvName:  pvName,
+		pvcName: pvc.ObjectMeta.Name,
+		scName:  scName,
+		config:  pvConfigMap,
```

Comment:
How about `options` instead of `config`?
Since the name of this struct contains `Config` already.

Reply:
done. changed to options.